Add sub-pixel accuracy to 2D alignment

to266 opened the pull request:
resubmission of https://github.com/hyperspy/hyperspy/pull/1236 in order to just quicken things up.

With a test and supports both cross- and phase-correlations.

vidartf committed:
Add sub-pixel accuracy to `Signal.estimate_shift2D`

Added an argument `sub_pixel_factor` to all functions that estimate
shifts for 2D signals. This factors allows reasonably fast sub-pixel
accuracy registration of images.

vidartf committed:
Missing import + style

vidartf committed:
Merge remote-tracking branch 'hyperspy/RELEASE_next_minor' into ENH_subpixel_2Dshifts

to266 committed:
Merge remote-tracking branch 'upstream/RELEASE_next_minor' into ENH_subpixel_2Dshifts

to266 committed:
add phase-correlation to subpixel estimation, and a test

dnjohnstone commented:
Had a quick play with this and it seems ok - although it needs merging up with the current RELEASE_next_minor

to266 committed:
Merge remote-tracking branch 'upstream/RELEASE_next_minor' into ENH_subpixel_2Dshifts

francisco-dlp commented on hyperspy/tests/signal/test_2D_tools.py:
```diff
@@ -58,6 +58,12 @@ def test_align_subpix(self):
         # Compare by broadcasting
         np.testing.assert_allclose(s.data[4], s.data[0], rtol=1)
 
+    def test_estimate_subpix(self):
```
The test for the case when ``normalize_corr`` is ``True`` is missing.

francisco-dlp commented:
Other than the comment above, it looks good to me too.

dnjohnstone commented:
Could we get this in to v1.2 please? Looks basically done!

francisco-dlp commented:
It needs a test for phase correlation. Anybody is welcome to contribute it.

to266 commented:
I think when I tried to add the missing test I was not happy with the result - namely even with fine sub-sampling the results were not that much better and not necessarily converged towards the true (known) solution. 

In the end I attempted to debug what the heck was happening, but could not find fault in the actual process. I think it does the correct thing and without mistakes, it's just that the result is not what we'd expect

to266 committed:
Merge branch 'RELEASE_next_minor' into ENH_subpixel_2Dshifts

francisco-dlp commented:
What I meant is that [this line](https://coveralls.io/builds/9686196/source?filename=hyperspy%2F_signals%2Fsignal2d.py#L101) is not tested.

ericpre committed:
Merge remote-tracking branch 'upstream/RELEASE_next_minor' into ENH_subpixel_2Dshifts

# Conflicts:
#	setup.py

ericpre committed:
Add test and fix a bug with `estimate_image_shift` returning the wrong `shifts`.

ericpre committed:
Add tests and fix a bug with plotting correlation.

ericpre committed:
Take max of real part of correlation to avoid casting complex value to real and get ride of numpy warning.
Switch backend to `agg` in a test to avoid showing many figures and holding the test.

francisco-dlp commented:
LGTM, but it requires merging Rnm in.

ericpre committed:
Merge remote-tracking branch 'upstream/RELEASE_next_minor' into ENH_subpixel_2Dshifts

# Conflicts:
#	hyperspy/_signals/signal2d.py
